boards/hail/src/main.rs

```diff
@@ -466,5 +467,8 @@ pub unsafe fn reset_handler() {
         debug!("{:?}", err);
     });
 
-    board_kernel.kernel_loop(&hail, chip, Some(&hail.ipc), &main_loop_capability);
+    let scheduler =
+        components::sched::round_robin::RoundRobinComponent::new(board_kernel, &PROCESSES)
+            .finalize(components::rr_component_helper!(NUM_PROCS));
+    scheduler.kernel_loop(&hail, chip, Some(&hail.ipc), &main_loop_capability);
```

Comment:
I'm not sure about changing all of the schedulers. At the very least the title of this PR should change. But I'm a little wary of changing the scheduler out from under the board authors.

Reply:
I don't like the idea of leaving a bunch of boards sticking around on something that is the equivalent of the current scheduler, mostly because the current scheduler allows for apps to indefinitely starve other apps, which is something that we sort of claim to have designed Tock to prevent (e.g. the evil while(1) example).

If people feel strongly about this I can change it back, but I really believe that universal use of a round robin scheduler would a preferable default.

I changed the title for now.

Reply:
I think I'm okay with changing schedulers as I don't see it as a change in _intended_ functionality. I thought the "default scheduler" of Tock was always intended to be round robin with pre-emptions. I see this more conceptually as a bugfix than as change if anything.

Reply:
The current scheduler is still implemented after this PR, right? If we don't like it, then we shouldn't keep it.

I'm fine with just updating the PR title, but in general I do want to make sure we are being deliberate when changing settings that are delegated to main.rs.

Reply:
The priority scheduler in this PR no longer enforces timeslices -- it is a strict priority scheduler where the highest priority process will always be running (with the kernel thread -- aka bottom half interrupts and deferred calls -- being higher priority than all userspace processes). I think this is worth keeping because priority scheduling can be useful in some systems, and this scheduler does not pretend that all apps will be given a chance to run when that is not in fact the case.